src/cpu.rs

```diff
@@ -49,13 +49,36 @@ impl ARM7TDMI {
     pub fn execute_instruction(&mut self, instruction: Instruction) {
         match instruction {
             AddSub { opcode, operand, source_register, destination_register } => {
+                use opcode::OpcodeAddSub::*;
                 match opcode {
                     AddRegister => {
-                        self.register_instruction(operand,
+                        let register_operand = operand;
+                        self.register_instruction(register_operand,
```

Comment:
Skip variable binding and put it straight into the function? (less lines of code for the human to read!)